contracts/upgradeable_contracts/components/common/SelectorTokenGasLimitManager.sol

```diff
@@ -0,0 +1,113 @@
+pragma solidity 0.7.5;
+
+import "../../BasicAMBMediator.sol";
+
+/**
+ * @title SelectorTokenGasLimitManager
+ * @dev Basic storage and methods needed by mediators to interact with AMB bridge.
+ * @dev Functionality for determining the request gas limit for AMB execution based on the request selectors and bridged tokens.
+ */
+abstract contract SelectorTokenGasLimitManager is BasicAMBMediator {
+    bytes32 internal constant REQUEST_GAS_LIMIT = 0x2dfd6c9f781bb6bbb5369c114e949b69ebb440ef3d4dd6b2836225eb1dc3a2be; // keccak256(abi.encodePacked("requestGasLimit"))
+
+    /**
+     * @dev Sets the default gas limit to be used in the message execution by the AMB bridge on the other network.
+     * This value can't exceed the parameter maxGasPerTx defined on the AMB bridge.
+     * Only the owner can call this method.
+     * @param _gasLimit the gas limit fot the message execution.
+     */
+    function setRequestGasLimit(uint256 _gasLimit) external onlyOwner {
+        _setRequestGasLimit(0x00000000, address(0), _gasLimit);
+    }
+
+    /**
+     * @dev Sets the selector-specific gas limit to be used in the message execution by the AMB bridge on the other network.
+     * This value can't exceed the parameter maxGasPerTx defined on the AMB bridge.
+     * Only the owner can call this method.
+     * @param _selector method selector of the outgoing message payload.
+     * @param _gasLimit the gas limit fot the message execution.
+     */
+    function setRequestGasLimit(bytes4 _selector, uint256 _gasLimit) external onlyOwner {
```

Comment:
Does it make sense to introduce helper methods to setup limits for known selectors?

Reply:
Let's do it after the merge of `relayTokensAndCall` functionality, since it introduces new selectors for communication between mediators. Then, I think it makes sense to introduce a single function for setting gas limits for all 7 selectors:
* `deployAndHandleBridgedTokens`/`deployAndHandleBridgedTokensAndCall`
* `handleBridgedTokens`/`handleBridgedTokensAndCall`
* `handleNativeTokens`/`handleNativeTokensAndCall`
* `fixFailedMessage`

Reply:
I have merged the PR #5. Please introduce the changes